product-service/src/handlers/getProductsById.ts

```diff
@@ -0,0 +1,31 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+
+import { productsList } from '../mockProductsList';
+
+export const getProductsById: APIGatewayProxyHandler = async event => {
+  try {
+    const { productId } = event.pathParameters;
+
+    const productById: Product = productsList.find(product => product.id === productId);
+
+    return productById 
+        ? {
+            statusCode: 200,
+            headers: { 
+              'Access-Control-Allow-Origin': '*',
+              'Access-Control-Allow-Credentials': true 
+            },
+            body: JSON.stringify(productById)
+          } 
+          : {
+            statusCode: 404,
+            body: JSON.stringify({ error: `Product with id: ${productById} is not found` })
+          };
+
+  } catch (e) {
+    return {
+      statusCode: 500,
+      body: `There is an unexpected error ${ JSON.stringify(e) }`

```

Comment:
I suggest to send only error message, but not a full error. It's also good practice to log error to see the stack trace
`console.error(e)`

Reply:
@AndreyS55 Thanks, I've made changes